### PR TITLE
CBAPI-2019: added device processes example script

### PR DESCRIPTION
**Pull request by abowersox-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Tests have been added that prove the fix is effective or that the feature works.
- [x] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?
 https://jira.carbonblack.local/browse/CBAPI-2019

## Pull Request Description

Added an example script to show multiple process searches running asynchronously, one for each device.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Script has been run against prod02.

### CODE REVIEW

File: examples/enterprise_edr/device_processes.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import logging
+import concurrent.futures
+from cbc_sdk.helpers import eprint, build_cli_parser, get_cb_cloud_object
+from cbc_sdk.platform import Device
+from cbc_sdk.enterprise_edr import Process
+
+log = logging.getLogger(__name__)
+
+
+def main():
```
Comment (avanbrunt-cb) [inline examples/enterprise_edr/device_processes.py:23]: Can we add some comments around the usage pattern for the cli input

File: examples/enterprise_edr/device_processes.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    parser = build_cli_parser()
+
+    args = parser.parse_args()
+    cb = get_cb_cloud_object(args)
+
+    query = cb.select(Device).where('1')
```
Comment (avanbrunt-cb) [inline examples/enterprise_edr/device_processes.py:29]: Let's make this a query for active devices using "ACTIVE" for device_status

**Review by avanbrunt-cb — APPROVED** [review]
Looks great only two small pieces of feedback

File: examples/enterprise_edr/device_processes.py (hunk 2 above)
Comment (abowersox-cb) [reply examples/enterprise_edr/device_processes.py:29]: How do you write that? Is it just "device_status:ACTIVE"?

File: examples/enterprise_edr/device_processes.py (hunk 1 above)
Comment (abowersox-cb) [reply examples/enterprise_edr/device_processes.py:23]: Why? None of our other examples has those.

Comment (avanbrunt-cb) [reply examples/enterprise_edr/device_processes.py:23]: This script does not have any descriptions of what it does we need at a minimum some context for what this does so a developer can't better utilize the example. Our other examples should have descriptions too

File: examples/enterprise_edr/device_processes.py (hunk 2 above)
Comment (avanbrunt-cb) [reply examples/enterprise_edr/device_processes.py:29]: It's a criteria property https://developer.carbonblack.com/reference/carbon-black-cloud/platform/latest/devices-api/

Comment (avanbrunt-cb) [reply examples/enterprise_edr/device_processes.py:29]: https://github.com/carbonblack/carbon-black-cloud-sdk-python/blob/develop/src/cbc_sdk/platform/devices.py#L335

Comment (abowersox-cb) [reply examples/enterprise_edr/device_processes.py:29]: OK, I remember now.  Adding the set_status() to the query reduced the number of devices being queried from 116 to 15.

File: examples/enterprise_edr/device_processes.py (hunk 1 above)
Comment (abowersox-cb) [reply examples/enterprise_edr/device_processes.py:28]: I added a docstring to the script to explain what exactly it does.  I suggest that an additional story be created to add those to other examples.

**Review by llyon-cb — APPROVED** [review]

**Pull request merged**